lib/apiservers/engine/backends/backends.go

```diff
@@ -93,44 +94,86 @@ func Init(portLayerAddr, product string, config *config.VirtualContainerHostConf
 	// the vic-machine installer timeout will intervene if this blocks for too long
 	pingPortLayer()
 
+	if err := hydrateCaches(); err != nil {
+		return err
+	}
+
 	log.Info("Creating image store")
 	if err := createImageStore(); err != nil {
 		log.Errorf("Failed to create image store")
 		return err
 	}
 
+	serviceOptions := registry.ServiceOptions{}
+	for _, registry := range insecureRegs {
+		insecureRegistries = append(insecureRegistries, registry.Path)
+	}
+	if len(insecureRegistries) > 0 {
+		serviceOptions.InsecureRegistries = insecureRegistries
+	}
+	log.Debugf("New registry service with options %#v", serviceOptions)
+	RegistryService = registry.NewService(serviceOptions)
+
+	return nil
+}
+
+func hydrateCaches() error {
+	wg := sync.WaitGroup{}
+	wg.Add(4)
```

Comment:
seems like a good place to ask this question as well. Is there any chance that any of the `wg.Done()` might not get called? Is there a way to attach timeouts to some of those goroutines just in case they get stuck?